### ISSUE TITLE
spring boot app can not startup in kubernetes ,the log is only printed a little of entire

### ISSUE THREAD

**Issue by 13567436138** [kubernetes/kubernetes#63396]
```
[root@node1 data-transfer]# kubectl logs data-transfer-web-8596955dc9-mdcdw 

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::        (v1.5.7.RELEASE)

2018-05-03 10:24:03.545  INFO 1 --- [           main] com.youben.Application                   : Starting Application on data-transfer-web-8596955dc9-mdcdw with PID 1 (/app.war started by root in /)
2018-05-03 10:24:03.646  INFO 1 --- [           main] com.youben.Application                   : No active profile set, falling back to default profiles: default
2018-05-03 10:24:06.045  INFO 1 --- [           main] ationConfigEmbeddedWebApplicationContext : Refreshing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@45ff54e6: startup date [Thu May 03 10:24:05 UTC 2018]; root of context hierarchy
[root@node1 data-transfer]# ^C
```

```
[root@node1 data-transfer]# cat data-transfer-deploy.yaml 
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: data-transfer-web
  labels:
    app: data-transfer
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: data-transfer
    spec:
      terminationGracePeriodSeconds: 60
      nodeSelector:
        deploy: app
      containers:
      - name: data-transfer-web
        image: 192.168.1.225:5000/youben/data-transfer:1.0-SNAPSHOT
        command: ["java"]
        args: ["-Djava.security.egd=file:/dev/./urandom","-jar","/app.war"]
        readinessProbe:
          httpGet:
            path: /boot/admins/indexes/index
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 60
          timeoutSeconds: 5
        livenessProbe:
          httpGet:
            path: /boot/admins/indexes/index
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 60
          timeoutSeconds: 5
        ports:
        - containerPort: 8080
        resources:
          requests:
            cpu: 100m
            memory: 100Mi
          limits:
            cpu: 200m
            memory: 500Mi
---
apiVersion: v1
kind: Service
metadata:
  name: data-transfer-web
  labels:
    app: data-transfer
spec:
  ports:
  - port: 8080
    targetPort: 8080
  selector:
    app: data-transfer
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: data-transfer-ingress
  labels: 
    app: data-transfer
spec:
  rules:
  - host: www.data-transfer.demo
    http:
      paths:
      - backend:
          serviceName: data-transfer-web
          servicePort: 8080
[root@node1 data-transfer]# 
```

[root@node1 data-transfer]# kubectl get pod
NAME                                            READY     STATUS    RESTARTS   AGE
data-transfer-web-8596955dc9-mdcdw              0/1       Running   20         1h
nginx-ingress-controller-658567f55-95fcx        1/1       Running   0          2h
nginx-ingress-default-backend-65fb4fd97-wvjnj   1/1       Running   0          2h

**Comment by dims** [comment]
if you are trying to see what's wrong in your app, then another thing you can try is to use `docker logs` on the container directly. 

/close

**Comment by 13567436138** [comment]
docker logs also have very little logs

**Comment by avikjis27** [comment]
I am facing the same issue. The docker log and pods' log are not complete.

**Comment by LakshyaGarg** [comment]
I am facing the same issue, spring boot app status shows CrashLoopBackOff and logs are not complete. Application does not start, what could be the solution ?

**Comment by am7489** [comment]
I am having the same issue. Has anyone found out what their problem was?  Not finding any apparent errors in logs. My docker image works locally deployed but on kubernetes the Spring Boot app won't fully start.

**Comment by LakshyaGarg** [comment]
I found the issue with my setup, as i had issue with my dns network in the cluster. Due to which my services were not able to interact.

**Comment by gpjacobs** [comment]
I was experiencing this same issue but found a different cause - the CPU resource limit. With a low CPU limit, the Spring Boot app was only logging a few lines (as shown above) and appeared to be stuck, and eventually the pod would be killed for failing the liveness/readiness probes. I kept increasing the CPU limit until it started up successfully. I needed to increase my probe delays a bit too. My guess is the Spring Boot bootstrapping process is very CPU intensive and when throttled, causes significant downstream delays.

Startup times for comparison:
(Baseline) Starting the app on my laptop, no container: ~10s
Starting the app in my K8s cluster with CPU limit of 100m: stalled/killed (failed liveness check)
Starting the app in my K8s cluster with CPU limit of 200m: stalled/killed (failed liveness check)
Starting the app in my K8s cluster with CPU limit of 300m: 40s
Starting the app in my K8s cluster with no CPU limit: 5s
